jobserver/views/status.py

```diff
@@ -1,20 +1,40 @@
 from django.db.models import Count
 from django.db.models.functions import Lower
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
 from django.template.response import TemplateResponse
+from django.utils import timezone
 from django.views.generic import View
 
 from ..backends import show_warning
 from ..models import Backend
 
 
-class Status(View):
+class PerBackendStatus(View):
     def get(self, request, *args, **kwargs):
-        def format_last_seen(last_seen):
-            if last_seen is None:
-                return "never"
+        backend = get_object_or_404(Backend, slug=self.kwargs["backend"])
+
+        try:
+            last_seen = backend.stats.order_by("-api_last_seen").first().api_last_seen
+        except AttributeError:
+            # don't show Backends which have never checked in as an error
+            last_seen = timezone.now()
+
+        # how long ago did we last see this backend?
+        time_since_last_seen = timezone.now() - last_seen
+
+        # do we consider the backend missing?
+        is_missing = time_since_last_seen > backend.alert_timeout
 
-            return last_seen.strftime("%Y-%m-%d %H:%M:%S")
+        # set a status code of 503 if the backend was last seen longer than the
+        # configured value ago
+        status_code = 503 if is_missing else 200
```

Comment:
Suggestion: return a text/plain response with `last_seen` as isodate

Freshping only needs the status code, but it's useful to have something there if we hit it with curl to test stuff, and the actual last seen value might be useful

Reply:
ooh yes 👍